source/compiler/qsc_qasm/src/compiler.rs

```diff
@@ -1114,6 +1133,20 @@ impl QasmCompiler {
             (PragmaKind::QdkBoxOpen | PragmaKind::QdkBoxClose, None) => {
                 self.push_compiler_error(CompilerErrorKind::MissingBoxPragmaTarget(stmt.span));
             }
+            (PragmaKind::QdkQirProfile, Some(profile)) => {
+                if Profile::from_str(profile).is_ok() {
+                    self.pragma_config
+                        .insert(PragmaKind::QdkQirProfile, profile.clone());
+                    return;
+                }
+                self.push_compiler_error(CompilerErrorKind::InvalidProfilePragmaTarget(
+                    profile.to_string(),
+                    stmt.value_span.unwrap_or(stmt.span),
+                ));
+            }
+            (PragmaKind::QdkQirProfile, None) => {
+                self.push_compiler_error(CompilerErrorKind::MissingProfilePragmaTarget(stmt.span));
+            }
```

Comment:
Are these really different cases? When a user is typing they will get one message for a moment, then the other. I think missing value is just a subcase of invalid value.

Reply:
They definitely could be merged, but I was trying to follow the pattern for the box pragma. I can see a notable difference between the two different pragmas though and why it makes more sense to have a separate error for invalid vs missing for box pragma, but not for profile pragma. I'll merge the two errors together.

Reply:
I've merged the two error kinds and tried to rephrase the error message to work better for both cases.